pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -334,6 +334,9 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr config.Config, event m
 
 // WorkloadInstanceHandler defines the handler for service instances generated by other registries
 func (s *ServiceEntryStore) WorkloadInstanceHandler(wi *model.WorkloadInstance, event model.Event) {
+	if !wi.Ready && event != model.EventDelete {
```

Comment:
what if a pod goes from ready to not ready? it would be an update event

Reply:
maybe want something like https://github.com/istio/istio/pull/29426/files#diff-dc676371f55a5124776027ab17fe30498e08d47a3cc206321f005ddab8747093R120?